src/widgets/mod.rs

```diff
@@ -59,7 +62,37 @@ bitflags! {
         /// Show the left border
         const LEFT   = 0b1000;
         /// Show all borders
-        const ALL = Self::TOP.bits | Self::RIGHT.bits | Self::BOTTOM.bits | Self::LEFT.bits;
+        const ALL = Self::TOP.bits() | Self::RIGHT.bits() | Self::BOTTOM.bits() | Self::LEFT.bits();
+    }
+}
+
+/// Implement the `Debug` trait for the `Borders` bitflags. This is a manual implementation to
+/// display the flags in a more readable way. The default implementation would display the
+/// flags as 'Border(0x0)' for `Borders::NONE` for example.
+impl Debug for Borders {
+    /// Display the Borders bitflags as a list of names. For example, `Borders::NONE` will be
+    /// displayed as `NONE` and `Borders::ALL` will be displayed as `ALL`. If multiple flags are
+    /// set, they will be displayed separated by a pipe character.
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_empty() {
+            return write!(f, "NONE");
+        }
+        if self.is_all() {
+            return write!(f, "ALL");
+        }
+        let mut first = true;
+        for (name, border) in self.iter_names() {
+            if border == Borders::NONE {
+                continue;
+            }
+            if first {
+                write!(f, "{}", name)?;
+                first = false;
+            } else {
+                write!(f, " | {}", name)?;
+            }
+        }
```

Comment:
```suggestion
        for (i, (name, border)) in self.iter_names().enumerate() {
            if border == Borders::NONE {
                continue;
            }
            if i == 0 {
                write!(f, "{}", name)?;
            } else {
                write!(f, " | {}", name)?;
            }
        }
```

Reply:
I actually wrote that approach and then switched to the version in this commit because there doesn't seem like an obvious guarantee that Borders::NONE will always be the first element. If bitflags changes to alpha order, or highest to lowest bit order then this fails. Maybe I'm too pessimistic on that though and we'd just catch that in tests and so it's not really a problem.

Reply:
Oh I see, it makes sense then.